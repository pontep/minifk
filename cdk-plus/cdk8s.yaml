language: typescript
app: node main.js
imports:
  - k8s
  - https://raw.githubusercontent.com/jenkinsci/kubernetes-operator/master/deploy/crds/jenkins.io_jenkins_crd.yaml
  - mattermost:=mattermost_crd.yaml
  - version.yaml
  - https://github.com/knative/serving/releases/download/v0.14.0/serving-crds.yaml
