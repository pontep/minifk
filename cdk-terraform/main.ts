import { Construct } from 'constructs';
import { App, TerraformStack } from 'cdktf';
import { KubernetesProvider, Namespace, Deployment, Service, Pod } from './.gen/providers/kubernetes';

class KubeStack extends TerraformStack {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    new KubernetesProvider(this, 'kind', {});

    const exampleNamespace = new Namespace(this, 'tf-cdk-example', {
      metadata: [{
        name: 'tf-cdk-example'
      }]
    })

    const app = 'nginx-example'
    new Deployment(this, 'nginx-deployment', {
      metadata: [{
        name: app,
        namespace: exampleNamespace.metadata[0].name,
        labels: {
          app
        }
      }],
      spec: [{
        replicas: 2,
        selector: [{
          matchLabels: {
            app
          }
        }],
        template: [{
          metadata: [{
            labels: {
              app
            }
          }],
          spec: [{
            container: [{
              image: 'nginx:1.7.8',
              name: 'example',
              port: [{
                containerPort: 80
              }],
              resources: [{
                limits: [{
                  cpu: '0.5',
                  memory: '512Mi'
                }],
                requests: [{
                  cpu: '250m',
                  memory: '50Mi'
                }]
              }]
            }]
          }]
        }]
      }]
    })

    new Service(this, 'nginx-service', {
      metadata: [{
        name: app
      }],
      spec: [{
        selector: {
          app
        },
        port: [{
          nodePort: 30201,
          port: 80,
          targetPort: '80'
        }],
        type: 'NodePort'
      }]
    })

    new Pod(this, 'hello-pod', {
      metadata: [
        {
          name: 'hello',
          labels: {
            app: 'hello'
          }
        }
      ],
      spec: [
        {
          container: [
            {
              name: 'hello',
              image: 'paulbouwer/hello-kubernetes:1.8',
              port: [
                {
                  name: 'hello',
                  containerPort: 8080
                }
              ]
            }
          ]
        }]
    })

    new Service(this, 'hello-service', {
      metadata: [
        {
          name: 'hello',
          labels: {
            app: 'hello'
          }
        }
      ],
      spec: [
        {
          selector: {
            app: 'hello'
          },
          port: [
            {
              name: 'hello',
              port: 80,
              targetPort: '8080',
            }
          ],
          type: 'LoadBalancer'
        }
      ]
    })

  }
}

const app = new App();
new KubeStack(app, 'tf-cdk-example');
app.synth();