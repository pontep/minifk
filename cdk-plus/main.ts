import { Construct } from 'constructs';
import { App, Chart } from 'cdk8s';
import { BookInfoGateWay } from './bookinfo/bookinfo-gateway';
import { CustomDestinationRule } from './bookinfo/custom-destination-rule';
import { BookInfo, BookInfoType } from './bookinfo/bookinfo'
// import { DetailsService } from './bookinfo/details'
export class Cdk8sPlusChart extends Chart {
    constructor(scope: Construct, id: string) {
        super(scope, id);

        new CustomDestinationRule(this, 'destination-rule', {})
        new BookInfoGateWay(this, 'bookinfo-gateway')

        // ส่งเลขเวอร์ชั่นให้ command -> รอบนี้จะ gen version 2,3, ..
        new BookInfo(this, 'details', {
            type: BookInfoType.DETAILS
        })
        new BookInfo(this, 'ratings', {
            type: BookInfoType.RATINGS
        })
        new BookInfo(this, 'reviews', {
            type: BookInfoType.REVIEWS
        })
        new BookInfo(this, 'reviews-v2', {
            type: BookInfoType.REVIEWS,
            version: "2"
        })
        // new BookInfo(this, 'reviews-v3', {
        //     type: BookInfoType.REVIEWS,
        //     version: "3"
        // })
        new BookInfo(this, 'productpage', {
            type: BookInfoType.PRODUCT_PAGE
        })

    }
}


export const app = new App();
new Cdk8sPlusChart(app, 'cdk8splus-chart');
app.synth();