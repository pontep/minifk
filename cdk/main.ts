import * as cdk8s from 'cdk8s';
import * as k8s from "./imports/k8s";

// our cdk app
const app = new cdk8s.App();

// our kuberentes chart
const chart = new cdk8s.Chart(app, 'cdk8s');

const labels = { app: 'hello-kubernetes', tier: 'frontend' };

new k8s.Deployment(chart, 'deployment', {
  spec: {
    selector: { matchLabels: labels },
    replicas: 5,
    template: {
      metadata: { labels },
      spec: {
        containers: [
          {
            name: 'hello-kubernetes',
            image: 'paulbouwer/hello-kubernetes:1.8',
            ports: [{ containerPort: 8080 }]
          }
        ]
      }
    }
  }
});

new k8s.Service(chart, 'service', {
  metadata: { labels },
  spec: {
    type: 'LoadBalancer',
    ports: [{ port: 80, targetPort: 8080 }],
    selector: labels,
  }
});

// we are done, synth
app.synth();