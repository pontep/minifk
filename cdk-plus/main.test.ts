import { Cdk8sPlusChart } from './main';
import { Testing } from "cdk8s";

describe('Test', () => {
    test('MatchSnapshot', () => {
        const app = Testing.app();
        const chart = new Cdk8sPlusChart(app, 'cdk8splus-chart');
        const results = Testing.synth(chart)
        expect(results).toMatchSnapshot();
    });
});
