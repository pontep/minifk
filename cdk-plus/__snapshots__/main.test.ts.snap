// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MatchSnapshot 1`] = `
Array [
  Object {
    "apiVersion": "networking.istio.io/v1alpha3",
    "kind": "DestinationRule",
    "metadata": Object {
      "name": "cdk8splus-chart-destination-rule-productpage-0c5dd4cd",
    },
    "spec": Object {
      "host": "productpage",
      "subsets": Array [
        Object {
          "labels": Object {
            "version": "v1",
          },
          "name": "v1",
        },
      ],
    },
  },
  Object {
    "apiVersion": "networking.istio.io/v1alpha3",
    "kind": "DestinationRule",
    "metadata": Object {
      "name": "cdk8splus-chart-destination-rule-reviews-15539156",
    },
    "spec": Object {
      "host": "reviews",
      "subsets": Array [
        Object {
          "labels": Object {
            "version": "v1",
          },
          "name": "v1",
        },
        Object {
          "labels": Object {
            "version": "v2",
          },
          "name": "v2",
        },
        Object {
          "labels": Object {
            "version": "v3",
          },
          "name": "v3",
        },
      ],
    },
  },
  Object {
    "apiVersion": "networking.istio.io/v1alpha3",
    "kind": "DestinationRule",
    "metadata": Object {
      "name": "cdk8splus-chart-destination-rule-ratings-5ac57045",
    },
    "spec": Object {
      "host": "ratings",
      "subsets": Array [
        Object {
          "labels": Object {
            "version": "v1",
          },
          "name": "v1",
        },
        Object {
          "labels": Object {
            "version": "v2",
          },
          "name": "v2",
        },
        Object {
          "labels": Object {
            "version": "v2-mysql",
          },
          "name": "v2-mysql",
        },
        Object {
          "labels": Object {
            "version": "v2-mysql-vm",
          },
          "name": "v2-mysql-vm",
        },
      ],
    },
  },
  Object {
    "apiVersion": "networking.istio.io/v1alpha3",
    "kind": "DestinationRule",
    "metadata": Object {
      "name": "cdk8splus-chart-destination-rule-details-c3afb86e",
    },
    "spec": Object {
      "host": "details",
      "subsets": Array [
        Object {
          "labels": Object {
            "version": "v1",
          },
          "name": "v1",
        },
        Object {
          "labels": Object {
            "version": "v2",
          },
          "name": "v2",
        },
      ],
    },
  },
  Object {
    "apiVersion": "networking.istio.io/v1alpha3",
    "kind": "Gateway",
    "metadata": Object {
      "name": "cdk8splus-chart-bookinfo-gateway-cd60aff0",
    },
    "spec": Object {
      "selector": Object {
        "istio": "ingressgateway",
      },
      "servers": Array [
        Object {
          "hosts": Array [
            "*",
          ],
          "port": Object {
            "name": "http",
            "number": 80,
            "protocol": "HTTP",
          },
        },
      ],
    },
  },
  Object {
    "apiVersion": "networking.istio.io/v1alpha3",
    "kind": "VirtualService",
    "metadata": Object {
      "name": "cdk8splus-chart-bookinfo-gateway-bookinfo-8cd6fbfd",
    },
    "spec": Object {
      "gateways": Array [
        "cdk8splus-chart-bookinfo-gateway-cd60aff0",
      ],
      "hosts": Array [
        "*",
      ],
      "http": Array [
        Object {
          "match": Array [
            Object {
              "uri": Object {
                "exact": "/productpage",
              },
            },
            Object {
              "uri": Object {
                "prefix": "/static",
              },
            },
            Object {
              "uri": Object {
                "exact": "/login",
              },
            },
            Object {
              "uri": Object {
                "exact": "/logout",
              },
            },
            Object {
              "uri": Object {
                "prefix": "/api/v1/products",
              },
            },
          ],
          "route": Array [
            Object {
              "destination": Object {
                "host": "productpage",
                "port": Object {
                  "number": 9080,
                },
              },
            },
          ],
        },
      ],
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "app": "details",
        "service": "details",
      },
      "name": "details",
    },
    "spec": Object {
      "externalIPs": Array [],
      "ports": Array [
        Object {
          "port": 9080,
        },
      ],
      "selector": Object {
        "app": "details",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "ServiceAccount",
    "metadata": Object {
      "labels": Object {
        "account": "details",
      },
      "name": "bookinfo-details",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "app": "details",
        "version": "v1",
      },
      "name": "details-v1",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "cdk8s.deployment": "cdk8spluschartdetailsdetailsv169834C29",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "details",
            "cdk8s.deployment": "cdk8spluschartdetailsdetailsv169834C29",
            "version": "v1",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [],
              "image": "docker.io/istio/examples-bookinfo-details-v1:1.15.1",
              "name": "details",
              "ports": Array [
                Object {
                  "containerPort": 9080,
                },
              ],
              "volumeMounts": Array [],
            },
          ],
          "serviceAccountName": "bookinfo-details",
          "volumes": Array [],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "app": "ratings",
        "service": "ratings",
      },
      "name": "ratings",
    },
    "spec": Object {
      "externalIPs": Array [],
      "ports": Array [
        Object {
          "port": 9080,
        },
      ],
      "selector": Object {
        "app": "ratings",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "ServiceAccount",
    "metadata": Object {
      "labels": Object {
        "account": "ratings",
      },
      "name": "bookinfo-ratings",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "app": "ratings",
        "version": "v1",
      },
      "name": "ratings-v1",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "cdk8s.deployment": "cdk8spluschartratingsratingsv1D30BC3B2",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "ratings",
            "cdk8s.deployment": "cdk8spluschartratingsratingsv1D30BC3B2",
            "version": "v1",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [],
              "image": "docker.io/istio/examples-bookinfo-ratings-v1:1.15.1",
              "name": "ratings",
              "ports": Array [
                Object {
                  "containerPort": 9080,
                },
              ],
              "volumeMounts": Array [],
            },
          ],
          "serviceAccountName": "bookinfo-ratings",
          "volumes": Array [],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "app": "reviews",
        "service": "reviews",
      },
      "name": "reviews",
    },
    "spec": Object {
      "externalIPs": Array [],
      "ports": Array [
        Object {
          "port": 9080,
        },
      ],
      "selector": Object {
        "app": "reviews",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "ServiceAccount",
    "metadata": Object {
      "labels": Object {
        "account": "reviews",
      },
      "name": "bookinfo-reviews",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "app": "reviews",
        "version": "v1",
      },
      "name": "reviews-v1",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "cdk8s.deployment": "cdk8spluschartreviewsreviewsv106C53748",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "reviews",
            "cdk8s.deployment": "cdk8spluschartreviewsreviewsv106C53748",
            "version": "v1",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "LOG_DIR",
                  "value": "/tmp/logs",
                },
              ],
              "image": "docker.io/istio/examples-bookinfo-reviews-v1:1.15.1",
              "name": "reviews",
              "ports": Array [
                Object {
                  "containerPort": 9080,
                },
              ],
              "volumeMounts": Array [
                Object {
                  "mountPath": "/tmp",
                  "name": "tmp",
                },
                Object {
                  "mountPath": "/opt/ibm/wlp/output",
                  "name": "wlp-output",
                },
              ],
            },
          ],
          "serviceAccountName": "bookinfo-reviews",
          "volumes": Array [
            Object {
              "emptyDir": Object {},
              "name": "tmp",
            },
            Object {
              "emptyDir": Object {},
              "name": "wlp-output",
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "app": "reviews",
        "service": "reviews",
      },
      "name": "reviews",
    },
    "spec": Object {
      "externalIPs": Array [],
      "ports": Array [
        Object {
          "port": 9080,
        },
      ],
      "selector": Object {
        "app": "reviews",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "ServiceAccount",
    "metadata": Object {
      "labels": Object {
        "account": "reviews",
      },
      "name": "bookinfo-reviews",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "app": "reviews",
        "version": "v2",
      },
      "name": "reviews-v2",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "cdk8s.deployment": "cdk8spluschartreviewsv2D1C78EC5",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "reviews",
            "cdk8s.deployment": "cdk8spluschartreviewsv2D1C78EC5",
            "version": "v2",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "LOG_DIR",
                  "value": "/tmp/logs",
                },
              ],
              "image": "docker.io/istio/examples-bookinfo-reviews-v2:1.15.1",
              "name": "reviews",
              "ports": Array [
                Object {
                  "containerPort": 9080,
                },
              ],
              "volumeMounts": Array [
                Object {
                  "mountPath": "/tmp",
                  "name": "tmp",
                },
                Object {
                  "mountPath": "/opt/ibm/wlp/output",
                  "name": "wlp-output",
                },
              ],
            },
          ],
          "serviceAccountName": "bookinfo-reviews",
          "volumes": Array [
            Object {
              "emptyDir": Object {},
              "name": "tmp",
            },
            Object {
              "emptyDir": Object {},
              "name": "wlp-output",
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "app": "reviews",
        "service": "reviews",
      },
      "name": "reviews",
    },
    "spec": Object {
      "externalIPs": Array [],
      "ports": Array [
        Object {
          "port": 9080,
        },
      ],
      "selector": Object {
        "app": "reviews",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "ServiceAccount",
    "metadata": Object {
      "labels": Object {
        "account": "reviews",
      },
      "name": "bookinfo-reviews",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "app": "reviews",
        "version": "v3",
      },
      "name": "reviews-v3",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "cdk8s.deployment": "cdk8spluschartreviewsv3AEEEF664",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "reviews",
            "cdk8s.deployment": "cdk8spluschartreviewsv3AEEEF664",
            "version": "v3",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "LOG_DIR",
                  "value": "/tmp/logs",
                },
              ],
              "image": "docker.io/istio/examples-bookinfo-reviews-v3:1.15.1",
              "name": "reviews",
              "ports": Array [
                Object {
                  "containerPort": 9080,
                },
              ],
              "volumeMounts": Array [
                Object {
                  "mountPath": "/tmp",
                  "name": "tmp",
                },
                Object {
                  "mountPath": "/opt/ibm/wlp/output",
                  "name": "wlp-output",
                },
              ],
            },
          ],
          "serviceAccountName": "bookinfo-reviews",
          "volumes": Array [
            Object {
              "emptyDir": Object {},
              "name": "tmp",
            },
            Object {
              "emptyDir": Object {},
              "name": "wlp-output",
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "app": "productpage",
        "service": "productpage",
      },
      "name": "productpage",
    },
    "spec": Object {
      "externalIPs": Array [],
      "ports": Array [
        Object {
          "port": 9080,
        },
      ],
      "selector": Object {
        "app": "productpage",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "ServiceAccount",
    "metadata": Object {
      "labels": Object {
        "account": "productpage",
      },
      "name": "bookinfo-productpage",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "app": "productpage",
        "version": "v1",
      },
      "name": "productpage-v1",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "cdk8s.deployment": "cdk8spluschartproductpageproductpagev11EFE0429",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "productpage",
            "cdk8s.deployment": "cdk8spluschartproductpageproductpagev11EFE0429",
            "version": "v1",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [],
              "image": "docker.io/istio/examples-bookinfo-productpage-v1:1.15.1",
              "name": "productpage",
              "ports": Array [
                Object {
                  "containerPort": 9080,
                },
              ],
              "volumeMounts": Array [
                Object {
                  "mountPath": "/tmp",
                  "name": "tmp",
                },
              ],
            },
          ],
          "serviceAccountName": "bookinfo-productpage",
          "volumes": Array [
            Object {
              "emptyDir": Object {},
              "name": "tmp",
            },
          ],
        },
      },
    },
  },
]
`;
